botocore/endpoint.py

```diff
@@ -227,6 +229,18 @@ def _get_response(self, request, operation_model, attempts):
                 request, verify=self.verify,
                 stream=operation_model.has_streaming_output,
                 proxies=self.proxies, timeout=self.timeout)
+        except ConnectionError as e:
+            # For a connection error, if it looks like it's a DNS
+            # lookup issue, 99% of the time this is due to a misconfigured
+            # region/endpoint so we'll raise a more specific error message
+            # to help users.
+            if self._looks_like_dns_error(e):
+                endpoint = e.request.url
```

Comment:
Considering this is inside the `Endpoint` class, we may want to call this something more specific, like `endpoint_url` or `request_url`.